File: microservice-architecture/distributed-tracing/js/test/test_tracing.js
import { childSpan, gatewayEntry } from '../src/tracing.js'

function assertEqual(expected, actual, message) {
  if (expected !== actual) {
    throw new Error(`${message} expected=${expected} actual=${actual}`)
  }
}

const gateway = gatewayEntry('TRACE-1001')
const order = childSpan(gateway, 'order-service', 'SPAN-ORDER')
const inventory = childSpan(order, 'inventory-service', 'SPAN-INVENTORY')

assertEqual(gateway.traceId, order.traceId, 'trace should reach order')
assertEqual(order.traceId, inventory.traceId, 'trace should reach inventory')
assertEqual(gateway.spanId, order.parentSpanId, 'order parent')
assertEqual(order.spanId, inventory.parentSpanId, 'inventory parent')

console.log('distributed-tracing(js) tests passed')

File: microservice-architecture/distributed-tracing/python/test/test_tracing.py
import os
import sys
import unittest

sys.path.append(os.getcwd())

from src.tracing import child_span, gateway_entry  # noqa: E402


class DistributedTracingTest(unittest.TestCase):
    def test_trace_propagation(self) -> None:
        gateway = gateway_entry("TRACE-1001")
        order = child_span(gateway, "order-service", "SPAN-ORDER")
        inventory = child_span(order, "inventory-service", "SPAN-INVENTORY")

        self.assertEqual(gateway.trace_id, order.trace_id)
        self.assertEqual(order.trace_id, inventory.trace_id)
        self.assertEqual(gateway.span_id, order.parent_span_id)
        self.assertEqual(order.span_id, inventory.parent_span_id)


if __name__ == "__main__":
    unittest.main()
